### PR TITLE
Minor enhancements to the GLTF module (lights and docs)

### PR DESCRIPTION
**Pull request by aaronfranke**
Minor misc enhancements to the GLTF module. Fix some minor things with lights, rename `Ref<GLTFNode> n` to `gltf_node`, give the extensions folder its own SCsub, and add some documentation.

### CODE REVIEW

**Review by fire — APPROVED** [review]
Split off from the other pr. Need to wait for cicd.

Edited: https://github.com/godotengine/godot/pull/66026

File: modules/gltf/SCsub [hunk 1]
```diff
@@ -7,7 +7,7 @@ env_gltf = env_modules.Clone()
 
 # Godot source files
 env_gltf.add_source_files(env.modules_sources, "*.cpp")
-env_gltf.add_source_files(env.modules_sources, "extensions/*.cpp")
 env_gltf.add_source_files(env.modules_sources, "structures/*.cpp")
+SConscript("extensions/SCsub")
```
Comment (akien-mga) [inline modules/gltf/SCsub:11]: What's the use for this? It's not done for structures nor editor, if there's no custom config to set for this folder / a sub env, I don't see how it makes things better.

Comment (aaronfranke) [reply modules/gltf/SCsub:11]: In isolation it's indeed not better, but it's planning for future PRs where there is more complex logic. In the future I'm planning to have `extensions/` to have subfolders in it. If you want I can remove it from this PR, and instead we can introduce this SCsub when it becomes useful.

Comment (akien-mga) [reply modules/gltf/SCsub:11]: I guess it's fine if you're confident that these future PRs will actually be merged as you envision them. But generally speaking yes, it's best to leave out such future proofing and do the change when it's actually useful.

**Comment by akien-mga** [comment]
I merged a second after you force pushed so it didn't work :)

Merged in 63c0dc6.

**Pull request closed without merging**